GeneratorInterface/GenFilters/plugins/PythiaFilterMotherSister.cc

```diff
@@ -66,17 +68,31 @@ bool PythiaFilterMotherSister::filter(edm::StreamID, edm::Event& iEvent, const e
                ++dau) {
             // find the daugther you're interested in
             if (abs((*dau)->pdg_id()) == abs(sisterID)) {
+              bool passTrackPt = false;
+              bool passLeptonPt = false;
+              // check pt of the nephews
+              for (HepMC::GenVertex::particle_iterator nephew = (*dau)->end_vertex()->particles_begin(HepMC::children);
+                   nephew != (*dau)->end_vertex()->particles_end(HepMC::children);
+                   ++nephew) {
+                int nephew_pdgId = abs((*nephew)->pdg_id());
+                // implicit requirement that only one newphew is a lepton
+                if (nephew_pdgId == 11 or nephew_pdgId == 13 or nephew_pdgId == 15)
+                  passLeptonPt = ((*nephew)->momentum().perp() > minLeptonPt);
+                if (nephew_pdgId == 211)
+                  passTrackPt = ((*nephew)->momentum().perp() > minTrackPt);
```

Comment:
What happens if you have multiple? With the current logic the last one need to be above. Is this what you had in mind? or do you mean "implicit assumption"?

Reply:
If the decay products of the sister particle include more than one lepton, only the last one will be checked. 
For my use case, this is irrelevant because by construction only one of the decay products will be a lepton.

Reply:
The code can be easily generalised to the case where multiple charged leptons/pions are present and all of them are required to pass the specified pt cuts.

Reply:
i would prefer moving to the generalized setup. @qliphy @silviodonato what do you think?

Reply:
A generalized setup is fine to me.

Reply:
Ok, thanks. The update will come tomorrow. We've had downtime at our t3 in the last two days.

Reply:
Thanks so much Maria!